examples/transactions.py

```diff
@@ -26,19 +21,32 @@ def __str__(self):
 
 
 async def run():
-    client = SqliteClient('example_transaction.sqlite3')
-    await client.create_connection()
-    Tortoise.init(client)
-    await generate_schema(client)
+    await Tortoise.init(config_file='config.json')
+    await Tortoise.generate_schemas()
 
     try:
-        async with client.in_transaction() as connection:
+        async with in_transaction() as connection:
             event = Event(name='Test')
             await event.save(using_db=connection)
             await Event.filter(id=event.id).using_db(connection).update(name='Updated name')
             saved_event = await Event.filter(name='Updated name').using_db(connection).first()
             await connection.execute_query('SELECT * FROM non_existent_table')
-    except sqlite3.OperationalError:
+    except OperationalError:
+        pass
+    saved_event = await Event.filter(name='Updated name').first()
+    print(saved_event)
+
+    @atomic()
```

Comment:
YAY!
I had lots of reservations about how transactions worked, as they were very non-intuive.

Is it possible to do `with atomic:`?

Reply:
You can do it with `async with in_transaction():` without calling it from connection. I separated methods because I am not sure it's intuitive to have function overloaded with two separate effects.